### PR TITLE
New/of real and sheduled cost (backport develop)

### PR DESCRIPTION
**Pull request by atm-jpb**
# NEW | Feature  add predicted_cost and real_cost on mrp_mo object

Add predicted cost and actual cost functionality. 
Add a caculateCostLines function in the mrp_mo object 
Add  a function that returns the unit price of a product 

Capture d’écran de 2022-08-16 17-45-05

we can see a  service condition in  calculatecostLines not implemented. i think this part can be handle after some other pr  in pipe that handle this particular point.

### CODE REVIEW

File: htdocs/mrp/class/mo.class.php [hunk 1]
```diff
@@ -1554,6 +1567,132 @@ public function getMoParent()
...
+		// foreach lines
+		if (is_array($this->lines) && count($this->lines)) {
+			require_once DOL_DOCUMENT_ROOT.'/product/class/product.class.php';
+			$tmpproduct = new Product($db);
+			$Tpredicted = array();
+			$Treal = array();
+			$totalRealCost = 0;
+			$totalPredictedCost = 0;
+
+			foreach ($this->lines as &$line) {
```
Comment (atm-lena) [inline htdocs/mrp/class/mo.class.php]: Vérifier que $this->lines n'est pas vide avant le foreach

File: htdocs/mrp/class/mo.class.php [hunk 2]
```diff
@@ -1554,6 +1567,132 @@ public function getMoParent()
...
+
+								if (!$Treal[$line->fk_product]) {
+									$Treal[$line->fk_product]['realCost'] = $productunitCost * $obj->Allqty;
+								}
+							}
+						}
+
+						// SERVICE
+					} elseif ($tmpproduct->type == $tmpproduct::TYPE_SERVICE) {
+						// This part is to be considered in a later development.
```
Comment (atm-lena) [inline htdocs/mrp/class/mo.class.php]: Peut-être mettre un //TODO ? Pour qu'il ressorte des comm' communs

File: htdocs/mrp/class/mo.class.php [hunk 3]
```diff
@@ -1554,6 +1567,132 @@ public function getMoParent()
...
+					}
+				}
+			}
+
+			foreach ($Tpredicted as $cost) {
+				$totalPredictedCost += $cost['predictedCost'];
+			}
+			foreach ($Treal as $cost) {
+				$totalRealCost += $cost['realCost'];
+			}
```
Comment (atm-lena) [inline htdocs/mrp/class/mo.class.php]: Vérif valeur vide avant foreach (les deux)

File: htdocs/mrp/class/mo.class.php (hunk 1 above)
Comment (atm-jpb) [reply htdocs/mrp/class/mo.class.php]: ok

File: htdocs/mrp/class/mo.class.php (hunk 2 above)
Comment (atm-jpb) [reply htdocs/mrp/class/mo.class.php]: ok

File: htdocs/mrp/class/mo.class.php (hunk 3 above)
Comment (atm-jpb) [reply htdocs/mrp/class/mo.class.php]: ok

**Pull request merged**